plugins/woocommerce-blocks/assets/js/editor-components/product-control/index.tsx

```diff
@@ -62,6 +62,16 @@ interface ProductControlProps {
 	 * Whether to show variations in the list of items available.
 	 */
 	showVariations?: boolean;
+	/**
+	 * Different messages to display in the component.
+	 * If any of the messages are not provided, the default message will be used.
+	 */
+	messages?: {
+		list?: string;
+		noItems?: string;
+		search?: string;
+		updated?: string;
```

Comment:
Can you explain what are those types of messages? I don't see any usage of `noItems` or `updated` 🤔

Reply:
As you can see here, `ProductControl` have [predefined messages](https://github.com/woocommerce/woocommerce/blob/be8d1259631c9c80fd67ced80de7bfb4eaff4169/plugins/woocommerce-blocks/assets/js/editor-components/product-control/index.tsx#L77-L82) i.e.
```tsx
const messages = {
	list: __( 'Products', 'woocommerce' ),
	noItems: __( "Your store doesn't have any products.", 'woocommerce' ),
	search: __( 'Search for a product to display', 'woocommerce' ),
	updated: __( 'Product search results updated.', 'woocommerce' ),
};
```

Therefore, I wanted to support same structure while making `ProductControl` block extensible. These messages are then passed to [SearchListControl](https://github.com/woocommerce/woocommerce/blob/be8d1259631c9c80fd67ced80de7bfb4eaff4169/plugins/woocommerce-blocks/assets/js/editor-components/product-control/index.tsx#L217-L234). 

Here is how [SearchListMessages](https://github.com/woocommerce/woocommerce/blob/be8d1259631c9c80fd67ced80de7bfb4eaff4169/plugins/woocommerce-blocks/assets/js/editor-components/search-list-control/types.ts#L58-L71) interface looks like:
```tsx
export interface SearchListMessages {
	// A more detailed label for the "Clear all" button, read to screen reader users.
	clear: string;
	// Message to display when the list is empty (implies nothing loaded from the server or parent component).
	noItems: string;
	// Message to display when no matching results are found. %s is the search term.
	noResults: string;
	// Label for the search input
	search: string;
	// Label for the selected items. This is actually a function, so that we can pass through the count of currently selected items.
	selected: ( n: number ) => string;
	// Label indicating that search results have changed, read to screen reader users.
	updated: string;
}
```

As you can see it supports both `noItems` and `updated` messages. 
- updated is used [here](https://github.com/woocommerce/woocommerce/blob/be8d1259631c9c80fd67ced80de7bfb4eaff4169/plugins/woocommerce-blocks/assets/js/editor-components/search-list-control/search-list-control.tsx#L228)
- noItems is used [here](https://github.com/woocommerce/woocommerce/blob/be8d1259631c9c80fd67ced80de7bfb4eaff4169/plugins/woocommerce-blocks/assets/js/editor-components/search-list-control/search-list-control.tsx#L169-L172)

Please let me know in case I am missing something. 🙇‍♂️

Reply:
Ah, TIL! Thanks <3